functions/New-DbaLinkedServer.ps1

```diff
@@ -0,0 +1,144 @@
+
+    .PARAMETER DataSource
+        The name of the data source.
+
+    .PARAMETER Location
+        The location of the database.
+
+    .PARAMETER ProviderString
+        The provider connection string.
+
+    .PARAMETER Catalog
+        The catalog or default database.
+
+    .PARAMETER InputObject
+        Allows piping from Connect-DbaInstance.
+
+    .PARAMETER WhatIf
+        Shows what would happen if the command were to run. No actions are actually performed.
+
+    .PARAMETER Confirm
+        Prompts you for confirmation before executing any changing operations within the command.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Security, Server
+        Author: Adam Lancaster https://github.com/lancasteradam
+
+        dbatools PowerShell module (https://dbatools.io)
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/New-DbaLinkedServer
+
+    .EXAMPLE
+        PS C:\>New-DbaLinkedServer -SqlInstance sql01 -LinkedServer linkedServer1 -ServerProduct mssql -Provider sqlncli -DataSource sql02
+
+        Creates a new linked server named linkedServer1 on the sql01 instance. The link is via the SQL Native Client and is connected to the sql02 instance.
+
+    .EXAMPLE
+        PS C:\>Connect-DbaInstance -SqlInstance sql01 | New-DbaLinkedServer -LinkedServer linkedServer1 -ServerProduct mssql -Provider sqlncli -DataSource sql02
+
+        Creates a new linked server named linkedServer1 on the sql01 instance. The link is via the SQL Native Client and is connected to the sql02 instance. The sql01 instance is passed in via pipeline.
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'Low')]
+    param (
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string]$LinkedServer,
+        [string]$ServerProduct,
+        [string]$Provider,
+        [string]$DataSource,
+        [string]$Location,
+        [string]$ProviderString,
+        [string]$Catalog,
+        [parameter(ValueFromPipeline)]
+        [Microsoft.SqlServer.Management.Smo.Server[]]$InputObject,
+        [switch]$EnableException
+    )
+    process {
+
+        if (Test-Bound -Not -ParameterName LinkedServer) {
+            Stop-Function -Message "LinkedServer is required"
+            return
+        }
+
+        foreach ($instance in $SqlInstance) {
+            $InputObject += Connect-DbaInstance -SqlInstance $instance -SqlCredential $SqlCredential
+        }
+
+        foreach ($instance in $InputObject) {
```

Comment:
If something is a server SMO with an active connection, so the output of Connect-DbaInstance, we call this $server. And there are definitly server SMOs in $InputObject so I would change this line to 
```
foreach ($server in $SqlInstance) {
```